storage/ndb/rest-server/data-access-rondb/src/rdrs-dal.cpp

```diff
@@ -104,15 +122,28 @@ RS_Status pk_read(RS_Buffer *reqBuff, RS_Buffer *respBuff) {
     return status;
   }
 
-  PKROperation pkread(reqBuff, respBuff, ndb_object);
-
-  status = pkread.PerformOperation();
+  Uint32 orid = OP_RETRY_INITIAL_DELAY;
+  Int32 orc   = OP_RETRY_COUNT;
+  do {
+    PKROperation pkread(reqBuff, respBuff, ndb_object);
+    status = pkread.PerformOperation();
+    if (OP_RETRY_COUNT != 0 && CanRetryOperation(status)) {
```

Comment:
Unless I first read `CanRetryOperation()`, I don't understand what's happening here. I would extract the http status 200 check first and do an early return. Only then does a check whether to retry make sense.